src/main/java/org/intellij/sonar/configuration/partials/SonarResourcesTableView.java

```diff
@@ -19,53 +23,51 @@
 import com.intellij.util.ui.ColumnInfo;
 import com.intellij.util.ui.ListTableModel;
 import org.intellij.sonar.configuration.ResourcesSelectionConfigurable;
+import org.intellij.sonar.persistence.Resource;
 import org.jetbrains.annotations.Nullable;
-import org.sonar.wsclient.services.Resource;
 
 public class SonarResourcesTableView {
 
   private final TableView<Resource> mySonarResourcesTable;
   private final JComponent myJComponent;
   private final SonarServersView mySonarServersView;
   private final Project myProject;
-  private static final ColumnInfo<Resource,String> TYPE_COLUMN = new ColumnInfo<Resource,String>("Type") {
+  private static final Map<String, String> TYPE_BY_QUALIFIER = ImmutableMap.<String, String>builder()
```

Comment:
private static final should be bbefore private final

it's a petty we don't have sonar analysis for the plugin itself ;)

Reply:
I didn't change that order. There were previously first the `private final`s, then the `private static final`s, I just added one. Please tell me whether you want me to move all around or stick to what it was. I just want to make sure. :-)

Reply:
Btw. just do analyze it. The guys from SonarSource provide a cloud instance with free usage for FOSS projects: https://sonarcloud.io

Reply:
oh did not know that, would be worth it to get a free sonar server

Reply:
nice works,travis:
https://travis-ci.org/sonar-intellij-plugin/sonar-intellij-plugin
sonarcloud:
https://sonarcloud.io/dashboard?id=sonar-intellij-plugin

Reply:
this will be now automatically updated if master is changed

Reply:
the "only" thing now missing is adjusting the rule set ;)

Reply:
alright, I defined a sonar-intellij-plugin-way rule set

Reply:
see
https://sonarcloud.io/organizations/sonar-intellij-plugin/quality_profiles/show?language=java&name=sonar-intellij-plugin-way

Reply:
And automatic preview testing of PRs with updating commit status in GitHub.

Reply:
So, what about the line move?

Reply:
> And automatic preview testing of PRs with updating commit status in GitHub.

how can I do this?

Reply:
> So, what about the line move?
the rule set did not complain so far
see for this file: https://sonarcloud.io/project/issues?fileUuids=AV8NYvGdVLHgQ8g1wbE6&id=sonar-intellij-plugin&resolved=false

Reply:
http://www.oracle.com/technetwork/java/codeconventions-150003.pdf
page 19
this is what I mean
don't know if there is a sonar rule for fields ordering

Reply:
hm travis analysis the pull requests also automatically:

https://travis-ci.org/sonar-intellij-plugin/sonar-intellij-plugin/pull_requests

maybe if you merge the changes from master the sonar analysis will update as well

Reply:
I know there is a checkstyle SQ rule that is capable of this.

Reply:
And not everyone follows the Oracle way of coding or not fully, that's why I replicated what was present.

Reply:
>I know there is a checkstyle SQ rule that is capable of this.
do you know the rule key?

Reply:
@Vampire if you register on sonarcloud I could add you to the plugins group, so you can do all the action by yourself ;)

Reply:
http://checkstyle.sourceforge.net/apidocs/com/puppycrawl/tools/checkstyle/checks/coding/DeclarationOrderCheck.html

Reply:
found the rule and it's acutally activated:

https://sonarcloud.io/organizations/sonar-intellij-plugin/rules#qprofile=AV8Nd7vPfwWpOfyf_jO0|activation=true|q=squid%3AS1213

Reply:
> .. it does not check the order of variables actually ...

The SQ own rule does not, the CS rule you found does. But to use it, the CS plugin needs to be installed I think, dunno whether it is present.

Reply:
https://docs.sonarqube.org/plugins/servlet/mobile#content/view/5311422

Reply:
> the changes I made on the master branch are not in your fork/ pull requests

Done

Reply:
Travis build did not work. :-(

Reply:
hmm... works on master?

Reply:
It says it cannot contact SonarQube on localhost, which sounds correct, as the SQ instance probably is not running on localhost. Maybe you have to setup something special for PRs? Besides that, as mentioned above the GitHub plugin is present on SonarCloud, so you can enable the nice GitHub PR handling with updating status and making comments about findings. The CheckStyle plugin is not present, so the ordering by visibility can probably not be checked as long as SonarSource is used.

Reply:
To get the GH PR magic working it seems you need to add some logic to the build script or to a dedicated travis shell script that decides based on environment variables what to do and in case of a PR set some SQ properties for the GH plugin and for doing a preview analysis. You then can also do analysis for different branches and have the results in SonarCloud concurrently. I guess the default Travis support only tweaks the build of the master branch for SonarCloud server and thus on the PR run it fails, as a normal run there would then update the SC project and that usually is not what you want.

Reply:
ye I see, this will need special attention, alright will first release the plugin tomorrow

Reply:
What about the reordering, should I do it or not?

Reply:
up to you, I would do it

Reply:
Ok, done